src/System.Runtime.CompilerServices.Unsafe/src/include/netfx/coreassembly.h

```diff
@@ -0,0 +1,8 @@
+#define CORE_ASSEMBLY "mscorlib"
```

Comment:
Can you help me understand why we need this/what this does?

Reply:
S.R.CS.Unsafe is a bit of a different project compared to the rest of our libraries, because the code is in IL instead of C#. Because of this, it will get built using ildasm instead of csc.exe, and that means you can't add a reference to a library (for example mscorlib) by simply adding `<Reference Include="mscorlib" />` on the .ilproj. Instead you need to add a header file that will define what CORE_ASSEMBLY means, since all of the IL code on the actual implementation refers to the core assembly as CORE_ASSEMBLY, so by adding this header file, you are basically adding a reference of some sorts to the project without having to change the IL. All this said, I'm not an expert on this, I just mimicked  what we were doing for other  target frameworks.